arrow-test/src/main/kotlin/arrow/test/laws/ConcurrentLaws.kt

```diff
@@ -184,8 +184,14 @@ object ConcurrentLaws {
         val mvar = MVar(a, this@acquireBracketIsNotCancelable).bind()
         mvar.take().bind()
         val p = Promise.uncancelable<F, Unit>(this@acquireBracketIsNotCancelable).bind()
-        val task = p.complete(Unit).flatMap { mvar.put(b) }
-          .bracket(use = { never<Int>() }, release = { unit() })
+        val task = just(Unit).bracket(
+          use = {
+            mvar.put(b)
+              .flatMap { p.complete(Unit) }
+              .flatMap { never<Int>() }
+          },
+          release = { unit() }
+        )
```

Comment:
This change is incorrect, but I can see that the test was incorrectly changed when the RxJava instances were introduced.

This tests verifies that `acquire` is not cancelable, and that  `mvar.put` still happens after `cancel` is called.

`MVar` is a single element blocking `Queue`, and we start with a filled `MVar` so every put will semantically block until `take`.

So we create a filled `MVar`, and a `Promise` as a latch to make sure we cancel after acquire starts.

Once the `Promise` is  completed, and the `Fiber` is thus running  `acquire` blocking on `mvar.put` we cancel the `Fiber`.

We should now be able to take 2 values  from the `MVar`, the  value we originally created the `MVar` with and value that was `put` into the `MVar` *after* we tried canceled the operation.


So the fixed test looks like follows.

```kotlin
fun <F> Concurrent<F>.acquireBracketIsNotCancelable(EQ: Eq<Kind<F, Int>>, ctx: CoroutineContext) =
  forAll(Gen.int(), Gen.int()) { a, b ->
    fx.concurrent {
      val mvar = MVar(a, this@acquireBracketIsNotCancelable).bind() // Create filled MVar
      val p = Promise.uncancelable<F, Unit>(this@acquireBracketIsNotCancelable).bind() // Create startAcquire latch
      val task = p.complete(Unit).flatMap { mvar.put(b) } // When acquire starts complete latch, and put value in MVar
        .bracket(use = { never<Int>() }, release = { unit() }) // Never finish `use` and no-op `release`.
      val (_, cancel) = task.fork(ctx).bind() // Fork the operation returning a `join, cancel` op
      p.get().bind() // Wait until acquire started
      cancel.fork(ctx).bind() // Cancel  the forked operation
      mvar.take().bind() // Take the original value from the MVar
                              // <-- Acquire blocks on `mvar.put` until here
      mvar.take().bind() // Take the value placed in MVar from cancelled acquire
    }.equalUnderTheLaw(just(b), EQ)
  }
```

Reply:
Reaktive does not pass your variant. I see the same problem here. After `p.get().bind()` is executed, `cancel.fork(ctx).bind()` maight be executed BEFORE execution of `mvar.put(b)`. If this happens, then the whole chain is disposed and the `flatMap` callback will never be called. What do you think?